HeterogeneousCore/CUDACore/interface/CUDAESProduct.h

```diff
@@ -86,7 +84,7 @@ class CUDAESProduct {
 private:
   struct Item {
     mutable std::mutex m_mutex;
-    CMS_THREAD_GUARD(m_mutex) mutable std::unique_ptr<cuda::event_t> m_event;
+    CMS_THREAD_GUARD(m_mutex) mutable std::shared_ptr<cuda::event_t> m_event;
```

Comment:
why this becomes a `shared_ptr` ?

Reply:
Because the event comes from the `CUDAEventCache` (instead of creating one explicitly), and that returns a `shared_ptr`.